pkg/rancher-components/tsconfig.json

```diff
@@ -0,0 +1,48 @@
+{
```

Comment:
We should avoid creating duplicated configuration and import a default one.

Reply:
I think it makes sense to do so, but I've run into issues with extending `tsconfig.default.json` because it defines `paths`.. I might be missing something, but I'm not coming across an elegant fix other than:

1. Refactor `tsconfig.default.json` so it doesn't define paths and allow individual components to define their own paths for aliases like `@shell` 
2. Understand that importing from `@shell` is a temporary measure until get into a position to publish a utils library and refactor at a later date

I'm open to other suggestions..

Reply:
Not so sure which problem you faced, however I tried by replacing this content and everything seems working: build, dev, test, linting, IDE imports.

```json
{
  "extends": "../../tsconfig.default.json",
  "exclude": [
    "node_modules"
  ]
}
```

There's some issues when I run the `yarn build:lib` command, which I guess is what you were mentioning, that would need to exclude some paths. 

```
ERROR in /Users/gleo/projects/dashboard/node_modules/@types/express/index.d.ts(58,55):
58:55 Type 'Response<any, Record<string, any>>' does not satisfy the constraint 'ServerResponse'.
  Types of property 'req' are incompatible.
    Type 'Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>> | undefined' is not assignable to type 'IncomingMessage'.
      Type 'undefined' is not assignable to type 'IncomingMessage'.
    56 |      * This is a built-in middleware function in Express. It serves static files and is based on serve-static.
    57 |      */
  > 58 |     var static: serveStatic.RequestHandlerConstructor<Response>;
       |                                                       ^
    59 | 
    60 |     /**
    61 |      * This is a built-in middleware function in Express. It parses incoming requests with urlencoded payloads and is based on body-parser.
```

Reply:
That's correct, `build:lib` fails and that's the piece that needs to be working. I'll spend some time seeing if I can't come up with a solution, but I created #5996 in the event that this requires more effort..